### PR TITLE
chore: automate the creation of a release

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It adds a script to generate a release:

1. git tag for the root package (testcontainers-go)
2. git tag for each Go module (examples and modules). The value of the tag will be the module type (examples or modules), slash, module name, slash, version
3. bump the version files

For that, we only need to read the current version form the version.go file.

The script is fairly simple:
- reads the version from the version.go file, keeping the responsibility to the build engineer, that has to check the version is correct.
- lists all subdirs in the well-known locations (examples and modules), excluding the `_template` dir
- removes the trailing slash in the module directory
- removes a possibly existing git tag for the module, skipping errors if the tag does not exist
- creates the git tag in the defined version
- stashes git to use a clean state
- updates the version.go file with the next development version, and mkdocs.yml file with the current version
- creates a commit in the main branch with those two files, only
- un-stashes git
- it creates a request to pkg.go.dev to add the releases (see https://pkg.go.dev/about#adding-a-package)

## Why is it important?
In order for Go modules to appear at pkg.go.dev, we need a valid Git tag for the module, and this script allows to release the library in the proper manner, creating a tag for the library and for each Go module, and requesting the addition in pkg.go.dev

## How to test this PR

Create a release in dry-run mode:
```shell
$ ./scripts/release.sh
git tag v0.18.0
git tag examples/bigtable/v0.18.0
git tag examples/cockroachdb/v0.18.0
git tag examples/consul/v0.18.0
git tag examples/datastore/v0.18.0
git tag examples/firestore/v0.18.0
git tag examples/mongodb/v0.18.0
git tag examples/mysql/v0.18.0
git tag examples/nginx/v0.18.0
git tag examples/postgres/v0.18.0
git tag examples/pubsub/v0.18.0
git tag examples/pulsar/v0.18.0
git tag examples/redis/v0.18.0
git tag examples/spanner/v0.18.0
git tag examples/toxiproxy/v0.18.0
git tag modules/compose/v0.18.0
git tag modules/localstack/v0.18.0
git stash
git checkout main
Producing a minor bump of the version, from v0.18.0 to 0.19.0
sed "s/const Version = ".*"/const Version = "0.19.0"/g" /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/internal/version.go > /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/internal/version.go.tmp
mv /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/internal/version.go.tmp /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/internal/version.go
sed "s/latest_version: .*/latest_version: v0.18.0/g" /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/mkdocs.yml > /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/mkdocs.yml.tmp
mv /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/mkdocs.yml.tmp /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/mkdocs.yml
git add /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/internal/version.go
git add /Users/mdelapenya/sourcecode/src/github.com/testcontainers/testcontainers-go/mkdocs.yml
git commit -m chore: prepare for next minor development cycle (0.19.0)
git push origin main --tags
git unstash
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/bigtable/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/cockroachdb/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/consul/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/datastore/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/firestore/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/mongodb/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/mysql/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/nginx/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/postgres/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/pubsub/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/pulsar/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/redis/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/spanner/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/examples/toxiproxy/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/modules/compose/@v/v0.18.0
curl https://proxy.golang.org/github.com/testcontainers/testcontainers-go/modules/localstack/@v/v0.18.0
```

Expected outcomes:
- a git tag for the library is created, and the same semver version for each Go module is created
    - v0.18.0 for root module
    - modules/compose/v0.18.0 for compose module
    - examples/modgodb/v0.18.0 for example modules
- the internal/version.go file is modified to the next bump
- the mkdocs.yml file is modified to the current version

## Follow-ups
Let's add here anything related to the release, to automate it as much as possible

### CODE REVIEW

**Comment by mdelapenya** [comment]
Mmm we probably need to follow https://pkg.go.dev/about#adding-a-package:

>## Adding a package
>Data for the site is downloaded from [proxy.golang.org](https://proxy.golang.org/). We monitor the [Go Module Index](https://index.golang.org/index) regularly for new packages to add to pkg.go.dev. If you don’t see a package on pkg.go.dev, you can add it by doing one of the following:
>
>- Visiting that page on pkg.go.dev, and clicking the “Request” button. For example:
https://pkg.go.dev/example.com/my/module
>
> - Making a request to proxy.golang.org for the module version, to any endpoint specified by the [Module proxy protocol](https://pkg.go.dev/cmd/go/#hdr-Module_proxy_protocol). For example:
https://proxy.golang.org/example.com/my/module/@v/v1.0.0.info
>
> - Downloading the package via the [go command](https://pkg.go.dev/cmd/go/#hdr-Add_dependencies_to_current_module_and_install_them). For example:
GOPROXY=https://proxy.golang.org GO111MODULE=on go get example.com/my/module@v1.0.0

Also we need to handle removing modules:
> ## Removing a package
>If you would like to hide versions of a module on pkg.go.dev, as well as from the go command, you should retract them. Retracting a module version involves adding a retract directive to your go.mod file and publishing a new version. See the Go blog post [New module changes in Go 1.16](https://go.dev/blog/go116-module-changes#module-retraction) and the [modules reference](https://go.dev/ref/mod#go-mod-file-retract) for details.
>
>Note that it is possible to retract the latest version of a module; the modules reference link above includes an example. Also note that published versions cannot be reused or modified, and this includes retracted versions.
>
>If you cannot publish a new version with retractions because the source code repository or domain name is no longer accessible, or if you want to hide documentation for all current and future versions, you can [file a request](https://go.dev/s/pkgsite-package-removal) for the pkgsite team to hide your package documentation from pkg.go.dev. Note that the package will continue to be available via go get or go install unless its module is retracted.

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#
+# Usage: ./scripts/release.sh "v0.0.1"
+# where v0.0.1 is the tag to be created.
+#
+# It's possible to run the script in dry-run mode, which will print the commands that
+# would be executed, without actually executing them.
+#
+# Usage: DRY_RUN="true" ./scripts/release.sh "v0.0.1" 
+
+readonly DRY_RUN="${DRY_RUN:-false}"
```
Comment (HofmeisterAn) [inline scripts/release.sh:15]: Should be the default `DRY_RUN` value `true`?

**Review by HofmeisterAn — DISMISSED** [review]

File: scripts/release.sh (hunk 1 above)
Comment (mdelapenya) [reply scripts/release.sh:15]: Indeed. Let me change it now

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+             "${directory}/${module_name}/${version}", e.g. "examples/mysql/v0.18.0", "modules/compose/v0.18.0"
+
+- The script will update the [mkdocs.yml](./mkdocks.yml) file, updating the `latest_version` field to the current version.
+- The script will update the [version.go](./internal/version.go) file, setting the next development version to the next **minor** version by default. For example, if the current version is `v0.18.0`, the script will update the [version.go](./internal/version.go) file with the next development version `v0.19.0`.
+- You can define the bump type, using the `BUMP_TYPE` environment variable. The default value is `minor`, but you can also use `major` or `patch` (the script will fail if the value is not one of these three):
+
+        BUMP_TYPE="major" ./scripts/release.sh
+
+- The script will create a commit in the **main** branch.
+- The script will push the git the main branch including the tags to the upstream repository, https://github.com/testcontainers/testcontainers-go
```
Comment (kiview) [inline RELEASING.md:26]: Let's switch those 2 lines around, since the tag push happens before the commit of the upcoming version.

**Review by kiview — APPROVED** [review]

File: RELEASING.md (hunk 2 above)
Comment (mdelapenya) [reply RELEASING.md:26]: L26 is pushing the git tags, not creating them

**Pull request merged**